tests/e2e/system/conftest.py

```diff
@@ -0,0 +1,420 @@
+            "Failed to change the parameter persistentVolumeReclaimPolicy"
+            f" to Delete {restored_noobaa_pv_name}"
+        )
+        logger.info(
+            "Changed the parameter persistentVolumeReclaimPolicy to Delete again"
+        )
+
+    def finalizer():
+        # Get the statefulset replica count
+        sst_obj = OCP(
+            kind=constants.STATEFULSET,
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        )
+        noobaa_db_sst_obj = sst_obj.get(resource_name=constants.NOOBAA_DB_STATEFULSET)
+        if noobaa_db_sst_obj["spec"]["replicas"] != 1:
+            modify_statefulset_replica_count(
+                statefulset_name=constants.NOOBAA_DB_STATEFULSET, replica_count=1
+            ), f"Failed to scale up the statefulset {constants.NOOBAA_DB_STATEFULSET}"
+
+        try:
+            restore_pvc_objs[0].delete()
+        except CommandFailed as ex:
+            if f'"{restore_pvc_objs[0].name}" not found' not in str(ex):
+                raise ex
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@pytest.fixture()
+def setup_mcg_system(
+    request,
+    awscli_pod_session,
+    mcg_obj_session,
+    bucket_factory,
+    cld_mgr,
+    test_directory_setup,
+):
+    # E2E TODO: Have a cluster with FIPS, KMS for RGW and Hugepages enabled
+    # E2E TODO: Please add the necessary skips to verify that all prerequisites are met
+
+    def mcg_system_setup(bucket_amount=5, object_amount=10):
+        # Create standard MCG buckets
+        test_buckets = bucket_factory(
+            amount=bucket_amount,
+            interface="CLI",
+        )
+
+        uploaded_objects_dir = test_directory_setup.origin_dir
+        downloaded_obejcts_dir = test_directory_setup.result_dir
+
+        test_buckets_pattern = "RandomObject-"
+        first_bidirectional_pattern = "FirstBidi-"
+        second_bidirectional_pattern = "SecondBidi-"
+        cache_pattern = "Cache-"
+
+        # Perform a round-trip object verification -
+        # 1. Generate random objects in uploaded_objects_dir
+        # 2. Upload the objects to the bucket
+        # 3. Download the objects from the bucket
+        # 4. Compare the object checksums in downloaded_obejcts_dir
+        # with the ones in uploaded_objects_dir
+        for count, bucket in enumerate(test_buckets):
+            assert random_object_round_trip_verification(
+                io_pod=awscli_pod_session,
+                bucket_name=bucket.name,
+                upload_dir=uploaded_objects_dir + f"Bucket{count}",
+                download_dir=downloaded_obejcts_dir + f"Bucket{count}",
+                amount=object_amount,
+                pattern=test_buckets_pattern,
+                mcg_obj=mcg_obj_session,
+            ), "Some or all written objects were not found in the list of downloaded objects"
+
+        # E2E TODO: Create RGW kafka notification & see the objects are notified to kafka
```

Comment:
Do we have the code for this? If there is a plan to merge this pr without it (since no code available), then please create a github issue to track it

Reply:
I will add kafka code on another PR

Reply:
Please create a github issue for adding Kafka code

Reply:
https://github.com/red-hat-storage/ocs-ci/issues/5274 <- github issue